### PR TITLE
Attempt to bring spacebones light up to speed with recent features

### PR DESCRIPTION
**Pull request by EpocSquadron**
* Adds cursorline (primary and secondary)
* Reworks colors slightly to try and match upstream theme better
* Distinguishes primary from regular selections
* Implements some of the markup specific styles
* Implements mode colors support

The upstream doesn't have much in the way of guides, so I did my best. Not expecting everyone to agree on the color choices but those can be further adjusted before or after merge.

### CODE REVIEW

File: runtime/themes/spacebones_light.toml [hunk 1]
```diff
@@ -24,57 +24,77 @@
...
+"ui.selection" = { bg = "hl2" }
+"ui.selection.primary" = { bg = "hl1" }
 "ui.cursor.primary" = { modifiers = ["reversed"] }
 "ui.cursor.match" = { bg = "bg3" }
+"ui.cursorline.primary" = { bg = "bg1" }
 "ui.menu" = { fg = "fg1", bg = "bg2" }
-"ui.menu.selected" = { fg = "#655370", bg = "#d1dcdf", modifiers = ["bold"] }
-"ui.virtual.whitespace" = "bg2"
+"ui.menu.selected" = { fg = "base", bg = "bg2", modifiers = ["bold"] }
+"ui.virtual" = "base-dim"
```
Comment (the-mikedavis) [inline runtime/themes/spacebones_light.toml:72]: The ruler is a little faint. Should it be something like

```toml
"ui.virtual.ruler" = { bg = "base-dim" }
```

so you can see it without having text cross it?

**Comment by the-mikedavis** [comment]
There's also `markup.heading.1`, `markup.heading.2` etc if you'd like to add those. It can be a bit loud to have so many colors for headings though so up to you

File: runtime/themes/spacebones_light.toml (hunk 1 above)
Comment (EpocSquadron) [reply runtime/themes/spacebones_light.toml:72]: I don't use the rulers feature, but trying it out I found `base-dim` to be too dark for my liking. I set it to `bg1` which is the same color as the cursorline, which makes it almost look like a crosshairs type thing. Not sure if that's acceptable, I can change it agian.

**Comment by EpocSquadron** [comment]
> There's also `markup.heading.1`, `markup.heading.2` etc if you'd like to add those. It can be a bit loud to have so many colors for headings though so up to you

Yeah I didn't feel the need to distinguish those colors and lacked enough guidance on what the colors should be -- the upstream theme has a fairly limited palette. I could look at the full palette of the spacemacs theme as that seems to be ultimately where this came from, but I'm not sure I want to put in that time when I can work towards some tree-sitter stuff I've been sitting on a long time.

**Review by the-mikedavis — APPROVED** [review]
Looks good, thanks for the updates! 🎨

Could you update the screenshot in the themes wiki?

**Pull request merged**